.tekton/scripts/check-partner-tasks.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$SCRIPTDIR/../.." || exit 1
+
+if [ ! -e "partners/" ]; then
+    echo "No partners directory exists. Skip checks against partner tasks."
+    exit 0
+fi
+
+check_dir_structure() {
+    resultf=$(mktemp)
+    check_result=$(mktemp)
+
+    for task_dir in partners/*; do
+        owners_file="$task_dir/OWNERS"
+        if [ ! -e "$owners_file" ]; then
+            echo "error: missing file $owners_file" >>"$check_result"
+            echo "issue found" >"$resultf"
+        fi
+
+        task_name=$(basename "$task_dir")
+        find "$task_dir" -mindepth 1 -maxdepth 1 -type d | \
+        while read -r version_dir; do
+            task_version=$(basename "$version_dir")
+            if ! echo "$task_version" | grep "^[0-9]\+\.[0-9]\+$" >/dev/null 2>&1; then
+                echo "error: $version_dir is not a version directory. A version directory name only contains digits and dot, e.g. 0.1, 1.2" >>"$check_result"
+                echo "issue found" >"$resultf"
+            else
+                # Ignore empty directory whatever it is a version directory or not.
+                if [ -n "$(ls "$version_dir")" ]; then
+                    found=
+                    # Check the task YAML file
+                    for yaml_file in "$version_dir"/*.yaml; do
+                        task_file=$(basename "$yaml_file")
+                        if [ "${task_name}.yaml" == "$task_file" ]; then
+                            found=1
+                            break
+                        fi
+                    done
+                    if [ -z "$found" ]; then
+                        echo "error: no task file is found under $version_dir. A task file must be named with the same task name. For example, task name is task1, then task file must have name task1.yaml" >>"$check_result"
+                        echo "issue found" >"$resultf"
+                    fi
+                    # Check README
+                    readme_file="$version_dir/README.md"
+                    if [ ! -s "$readme_file" ]; then
+                        echo "warning: it is recommended to provide an informative document for task $task_name in $readme_file" >>"$check_result"
+                    fi
+                fi
+            fi
+        done
+    done
+
+    if [ -s "$check_result" ]; then
+        cat "$check_result"
+        echo
+    fi
+
+    [ -s "$resultf" ] && return 1
+    return 0
+}
+
+check_privilege_use() {
+    resultf=$(mktemp)
+    check_result=$(mktemp)
+
+    check_cases="\
+allowPrivilegeEscalation:true:allowPrivilegeEscalation is not allowed to be true to request more privileges.
+runAsUser:0:root(0) is not allowed to be set to securityContext.runAsUser.
```

Comment:
currently some of our tasks are using runAsUser so it should be probably allowed. It's usually needed when PVC is used.